### PR TITLE
compiler: Misc code generation fixes

### PR DESCRIPTION
**Pull request by FabioLuporini**
Mostly emerged in the past month by developing the CUDA backend in devitopro

I've collected them here in this PR

PR equipped with tests (or updating the existing ones)

### CODE REVIEW

**Review by georgebisbas — COMMENTED** [review]
I had a very quick look, seems ok, but I will pass again tomorrow with more attention.

File: devito/passes/clusters/blocking.py [hunk 1]
```diff
@@ -338,12 +338,12 @@ def decompose(ispace, d, block_dims):
             intervals.append(i)
 
     # Create the intervals relations
-    # 1: `bbd > bd > d`
+    # 1: `bbd > bd`
```
Comment (georgebisbas) [inline devito/passes/clusters/blocking.py:341]: οκ

File: tests/test_dle.py [hunk 2]
```diff
@@ -193,6 +194,29 @@ def test_cache_blocking_structure_optrelax():
...
+    nbl = 3
+
+    damp = Function(name='damp', grid=grid)
+
+    eqns = [Eq(damp, 0.)]
+    for d in damp.dimensions:
+        # Left
+        dl = SubDimension.left(name='%sl' % d.name, parent=d, thickness=nbl)
+        eqns.extend([Inc(damp.subs({d: dl}), 1.)])
+        # right
```
Comment (georgebisbas) [inline tests/test_dle.py:208]: nitpicking right/left lowercase or capital for homogeneity

**Review by georgebisbas — APPROVED** [review]
Looks good to me!

File: devito/core/operator.py [hunk 3]
```diff
@@ -15,7 +16,77 @@
 
 class BasicOperator(Operator):
 
-    # The various Operator subclasses are expected to override the following attributes
+    # Default values for various optimization options
```
Comment (mloubout) [inline devito/core/operator.py:19]: yay!

File: devito/symbolics/inspection.py [hunk 4]
```diff
@@ -110,7 +110,8 @@ def estimate_cost(exprs, estimate=False):
 estimate_values = {
     'elementary': 100,
     'pow': 50,
-    'div': 5
+    'div': 5,
+    'Abs': 5,
```
Comment (mloubout) [inline devito/symbolics/inspection.py:114]: really? Isn't it a single byte change

**Review by mloubout — APPROVED** [review]
Minor comments, gtg

File: devito/symbolics/inspection.py (hunk 4 above)
Comment (FabioLuporini) [reply devito/symbolics/inspection.py:114]: so, before this change the associated cost would have been 100 (elementary) 😬 

if you google, you'll see there's no straightforward answer. Maybe with integers is like you say, but in general I'm not sure. After 60 secs of googling I gave up and decided to put 5. What really matters is that it's not 100 anymore

**Comment by FabioLuporini** [comment]
Merged

**Pull request merged**